Assembly-CSharp/Patches/AudioPlayerOneShotSingle.cs

```diff
@@ -0,0 +1,25 @@
+﻿using MonoMod;
+using System;
+using UnityEngine;
+
+// ReSharper disable All
+#pragma warning disable 1591, 0108, 0169, 0649, 114, 0414,0162, CS0626, IDE1005, IDE1006
+
+namespace Modding.Patches
+{
+    // These changes fix NREs that happen in this class when pre-processing scenes without a hero in them
+    [MonoModPatch("global::HutongGames.PlayMaker.Actions.AudioPlayerOneShotSingle")]
+    public class AudioPlayerOneShotSingle : global::HutongGames.PlayMaker.Actions.AudioPlayerOneShotSingle
+    {
+        private extern void orig_DoPlayRandomClip();
+        private void DoPlayRandomClip()
+        {
+            try
+            {
+                orig_DoPlayRandomClip();
+            }
+            catch (NullReferenceException)
```

Comment:
```suggestion
            catch (NullReferenceException) when (ModLoader.Preloading)
```

If you're going to catch NREs because of preloading, which seems like a good idea - you'd want to restrict it to only during the preloading itself. Otherwise mods which edit AudioPlayerOneShotSIngle and cause an NRE won't get any errors.

Reply:
Added this in latest commit, but since `ModLoader.Preloading` doesn't exist, i just use `!ModLoader.Preloaded`